src/main/java/services/RestoreFactory.java

```diff
@@ -351,6 +353,14 @@ private String getSyncToken(String username) {
         return storage.getMetaDataFieldForRecord(users.firstElement(), User.META_SYNC_TOKEN);
     }
 
+    // Device ID for tracking usage in the same way Android uses IMEI
+    private String getSyncDeviceId() {
+        if (asUsername == null) {
+            return DEVICE_ID_SLUG;
+        }
+        return String.format("%s|%s|as|%s", DEVICE_ID_SLUG, username, asUsername);
```

Comment:
Getting the `userId` would often require a SQLite DB lookup (since the other parameters are sent in the request) so I'd prefer not to incur that extra cost/point of failure unless necessary.